src/libraries/Common/src/Interop/OSX/Swift.Runtime/CryptoKit.cs

```diff
@@ -0,0 +1,274 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Swift;
+
+#pragma warning disable CS3016 // Arrays as attribute arguments are not CLS Compliant
+#pragma warning disable SYSLIB1051
+
+namespace Swift.Runtime
+{
+    /// <summary>
+    /// Represents ChaChaPoly in C#.
+    /// </summary>
+    internal unsafe partial struct ChaChaPoly
+    {
+        /// <summary>
+        /// Represents Nonce in C#.
+        /// </summary>
+        internal sealed unsafe partial class Nonce
+        {
+            private const int _payloadSize = 16;
+            internal void* payload;
+
+            internal Nonce()
+            {
+                payload = Marshal.AllocHGlobal(_payloadSize).ToPointer();
```

Comment:
Use `NativeMemory.Alloc`? It won't have a behavior difference but they deal with pointers and not intptrs so the code will be slightly cleaner.

Reply:
> they deal with pointers and not intptrs so the code will be slightly cleaner.

That is true, but not really the compelling reason for this one. The memory allocator's on the `Marshal` class should almost exclusively be used for cases on Windows. There are a few cases where using `Marshal. AllocCoTaskMem` is appropriate on non-Windows, but the other APIs (for example, `Marshal.AllocHGlobal`) should never appear in non-Windows code. Deferring to `NativeMemory.Alloc` should always be preferred except in a very few set of cases.

Reply:
Thanks, updated.